rfcs/text/0044-fileless-execution-linux.md

```diff
@@ -0,0 +1,132 @@
+# 0044: Fileless execution on Linux
+<!-- Leave this ID at 0000. The ECS team will assign a unique, contiguous RFC number upon merging the initial stage of this RFC. -->
+
+- Stage: **0 (strawperson)** <!-- Update to reflect target stage. See https://elastic.github.io/ecs/stages.html -->
+- Date: **2024-03-20** <!-- The ECS team sets this date at merge time. This is the date of the latest stage advancement. -->
+
+<!--
+As you work on your RFC, use the "Stage N" comments to guide you in what you should focus on, for the stage you're targeting.
+Feel free to remove these comments as you go along.
+-->
+
+<!--
+Stage 0: Provide a high level summary of the premise of these changes. Briefly describe the nature, purpose, and impact of the changes. ~2-5 sentences.
+-->
+
+This RFC proposes adding new fields and event types to enhance the detection of fileless malware execution and related malicious activities on Linux systems.
+
+The new fields include:
+ * file.is_memfd - Indicates if the file is an anonymous file descriptor (memfd) created using the memfd_create system call.
```

Comment:
There's already a `file.attributes` list. Maybe these could be added to the attributes instead of having separate fields?

Reply:
That could work, maybe in that case the attribute could be called just "memfd".